### ISSUE TITLE
maybe memory leak when self-play

### ISSUE THREAD

**Issue by gooooloo** [gooooloo/alpha-zero-in-python#4]

**Comment by apollo-time** [comment]
Please see my implements.
https://github.com/apollo-time/alpha-zero-in-python/blob/master/src/reversi_zero/agent/player.py
I modified your tree with synchronize.

**Comment by apollo-time** [comment]
And please see https://github.com/mokemokechicken/reversi-alpha-zero/issues/22

**Comment by gooooloo** [comment]
@apollo-time Yes you are right, there is multi-thread issue. Will fix according to your change. Thanks!
